clients/naming_client/host_reactor.go

```diff
@@ -171,3 +176,52 @@ func (hr *HostReactor) asyncUpdateService() {
 		time.Sleep(1 * time.Second)
 	}
 }
+
+// return true when service instance changed ,otherwise return false.
+func isServiceInstanceChanged(oldService, newService *model.Service) bool {
+	oldHostsLen := len(oldService.Hosts)
+	newHostsLen := len(newService.Hosts)
+	if oldHostsLen != newHostsLen {
+		return true
+	}
+	// compare refTime
+	oldRefTime := oldService.LastRefTime
+	newRefTime := newService.LastRefTime
+	if oldRefTime > newRefTime {
+		logger.Warn(fmt.Sprintf("out of date data received, old-t: %v , new-t:  %v", oldRefTime, newRefTime))
+		return false
+	}
+	// sort instance list
+	oldInstance := oldService.Hosts
+	newInstance := newService.Hosts
+	sortInstance(oldInstance)
+	sortInstance(newInstance)
```

Comment:
Sort operation will modify original order. Not sure will cause something wrong. 
Maybe we should copy, sort and compare these two hosts slices instead of breaking the original order.

Reply:
done